src/Engine/ProtoCore/Utils/CoreUtils.cs

```diff
@@ -765,18 +765,24 @@ public static string GetIdentifierExceptMethodName(ProtoCore.AST.AssociativeAST.
         /// Traverses the identifierlist argument until class name resolution succeeds or fails.
         /// </summary>
         /// <param name="classTable"></param>
-        /// <param name="identList"></param>
+        /// <param name="identifier"></param>
         /// <returns></returns>
-        public static string[] GetResolvedClassName(ClassTable classTable, IdentifierListNode identList)
+        public static string[] GetResolvedClassName(ClassTable classTable, AssociativeNode identifier)
         {
-            string[] classNames = classTable.GetAllMatchingClasses(GetIdentifierStringUntilFirstParenthesis(identList));
+            var identList = identifier as IdentifierListNode;
+            var identifierNode = identifier as IdentifierNode;
+            Validity.Assert(identList != null || identifierNode != null);
+
+            string partialName = identList != null ? GetIdentifierStringUntilFirstParenthesis(identList) : identifier.Name;
+
+            string[] classNames = classTable.GetAllMatchingClasses(partialName);
 
             // Failed to find the first time
             // Attempt to remove identifiers in the identifierlist until we find a class or not
             while (0 == classNames.Length)
             {
                 // Move to the left node
-                AssociativeNode leftNode = identList.LeftNode;
+                AssociativeNode leftNode = identList != null ? identList.LeftNode : identifierNode;
```

Comment:
I changed this API to accept either an `IdentifierListNode` or `IdentifierNode` from which to resolve class names. The `identList` can therefore be `null` if the input node is an `IdentifierNode`. Ideally it would have been nice to have a common base class for  `IdentifierListNode` and `IdentifierNode` since there are use cases such as this where both go hand-in-hand.